build: Fix target name

hebasto opened the pull request:
It seems like a typo :)
This PR:
- fixes errors when building a package in depends for `HOST=x86_64-apple-darwin16` (fix #19799)
- is a correct alternative to https://github.com/bitcoin/bitcoin/pull/19764/commits/d25e0e308f51f6b995e5d0033224c424dc8afc2c from #19764

hebasto committed:
build: Fix target name

hebasto commented:
cc @theuni @dongcarl @fanquake

hebasto commented:
cc @icota

icota commented:
I can confirm https://github.com/bitcoin/bitcoin/pull/19868/commits/3eb789c7ea9714cb7e70a7b06eb8abb9aaf33435 fixes https://github.com/bitcoin/bitcoin/issues/19799 but don't have the experience to weigh in if it's more correct than https://github.com/bitcoin/bitcoin/commit/d25e0e308f51f6b995e5d0033224c424dc8afc2c

laanwj commented:
Your PR description is a bit terse. Can you please explain why you think it was a typo, and why it is correct alternative to that other commit?

hebasto commented:
@laanwj 
> Your PR description is a bit terse. Can you please explain why you think it was a typo...

`$(package)_unpacked` targets are not used as prerequisites for other targets. It was naturally to admit that this line is a remnant from some previous state of work before it was merged.

> ... and why it is correct alternative to that other commit?

Because of the initial intention of this line that is expressed in the comment:
https://github.com/bitcoin/bitcoin/blob/3eb789c7ea9714cb7e70a7b06eb8abb9aaf33435/depends/funcs.mk#L267

theuni commented:
@hebasto is absolutely correct [here](https://github.com/bitcoin/bitcoin/pull/19868#issuecomment-688841314). Thanks for tracking this down!

From IRC Build meeting today:
```
<cfields> Heh, definitely a typo. Conceptually, that should be "_extracted".
<cfields> Well, I'm not sure if _extracted will actually work. But that's definitely what I intended.
<cfields> Conceptually it means: never start working on a target package (first step is extraction) before all build packages are done.
```

So Concept ACK, but I think it should be ```_extracted``` rather than ```_configured``` (assuming that works as intended).

hebasto commented:
Updated 3eb789c7ea9714cb7e70a7b06eb8abb9aaf33435 -> 7a89f2e6c539a54bcaa24bff41aae3910244ad3d ([pr19868.01](https://github.com/hebasto/bitcoin/commits/pr19868.01) -> [pr19868.02](https://github.com/hebasto/bitcoin/commits/pr19868.02), [diff](https://github.com/hebasto/bitcoin/compare/pr19868.01..pr19868.02)):

- addressed @theuni's [comment](https://github.com/bitcoin/bitcoin/pull/19868#issuecomment-693522171):
> So Concept ACK, but I think it should be `_extracted` rather than `_configured` (assuming that works as intended).

icota commented:
tACK https://github.com/bitcoin/bitcoin/pull/19868/commits/7a89f2e6c539a54bcaa24bff41aae3910244ad3d

dongcarl commented:
Code Review ACK 7a89f2e6c539a54bcaa24bff41aae3910244ad3d

theuni commented:
I was a bit hesitant to simply "fix" this typo, since the intended behavior has never actually executed. But it seems that the intended behavior was necessary and correct after all.

Lightly tested locally, corner cases like ```make download``` still work as expected.

ACK 7a89f2e6c539a54bcaa24bff41aae3910244ad3d.

fanquake commented:
Was wondering why the Windows builds seemed to fail here, but turns out the not just ran out of space:
```bash
+ find bitcoin-475d0ad3f464/bin -type f -executable -print0
/usr/bin/x86_64-w64-mingw32-objcopy:bitcoin-475d0ad3f464/bin/test_bitcoin.exe.dbg[.debug_info]: No space left on device
```
